---
title: Survival analysis with Risk Score from gene signatures
date: June 23, 2020
output: 
  prettydoc::html_pretty:
    theme: cayman
    highlight: github
---

## Idea

Big data experiments lead to big results. But what to do with a large gene signature from an experiment like [RNASeq](https://de.wikipedia.org/wiki/RNA-Seq)? In 2019, [Hess et al.](https://clincancerres.aacrjournals.org/content/25/5/1505.long) described a convenient way to filter out gene candidates from a large pool and compress it into a **Risk Score**. I applied this idea to a gene signature described by [Puram et al.](https://www.cell.com/cell/fulltext/S0092-8674(17)31270-9?_returnURL=https%3A%2F%2Flinkinghub.elsevier.com%2Fretrieve%2Fpii%2FS0092867417312709%3Fshowall%3Dtrue). 

## What did I do?

- Extract Puram gene signature
- Extract genes for [TCGA](https://www.cancer.gov/about-nci/organization/ccg/research/structural-genomics/tcga) patient cohort
- Use Cox PH models & [*rbsurv*](https://www.bioconductor.org/packages/release/bioc/vignettes/rbsurv/inst/doc/rbsurv.pdf) to identify genes
- Compress gene expression into Risk Score
- Quick survival analysis with [clinical data](https://www.cbioportal.org/study/summary?id=hnsc_tcga_pub) from TCGA


## Results

This is a glimpse on the actual analysis I did with the clinical data from TCGA. Take look.

```{r, fig.align='center', fig.cap=c("Correlation of all genes described by Puram et al.","Risk Score in accordance to Hess et al.","Testing performance of Risk Score in survival analysis"), echo=FALSE,message=F,warning=F}
# Load libraries
rm(list=ls())
knitr::opts_chunk$set(echo = F, message = F, warning = F)
'%!in%' <- function(x,y)!('%in%'(x,y))
library(cgdsr)
library(singscore)
library(tidyverse)
library(doBy)
library(survival)
library(survminer)
library(corrplot)
library(Hmisc)
library(ggpubr)
library(gtools)


setwd("~/Desktop/R/3_GitHub/RiskScore_pEMT/")
Puram <- t(read.csv(file = "Puram_pEMT_signature.csv", header = F, sep = ";", stringsAsFactors = F))
pemt.genes <- Puram[,2]
pemt.genes <- gsub(" ", "", pemt.genes, fixed = TRUE)

mycgds = CGDS("https://www.cbioportal.org/")
a <- getCancerStudies(mycgds)
mycancerstudy = getCancerStudies(mycgds)[105,1]
mycaselist = getCaseLists(mycgds,mycancerstudy)[1,1]
mygeneticprofile = getGeneticProfiles(mycgds,mycancerstudy)[3,1]
gen.prof <- getProfileData(mycgds,pemt.genes,mygeneticprofile,mycaselist)
found <- which(colnames(gen.prof)%in%pemt.genes)
gen.prof <- gen.prof[,found]

data.clin <-  readxl::read_xlsx("~/Desktop/R/3_GitHub/RiskScore_pEMT/TCGA_clinical.xlsx")
data.clin$ID <- data.clin$`SAMPLE ID`
data.clin$ID <- chartr('-', '.',data.clin$ID)
data.clin <- orderBy(~ID, data.clin)

gen.prof$ID <- rownames(gen.prof)
gen.prof <- orderBy(~ID, gen.prof)

data.clin$OS_MONTHS[data.clin$OS_MONTHS >= 60] <- 60
data.clin$OS_STATUS[data.clin$OS_MONTHS == 60 ] <- "LIVING"

data.clin$OS_STATUS[data.clin$OS_STATUS == "DECEASED"] <- 1
data.clin$OS_STATUS[data.clin$OS_STATUS == "LIVING"] <- 0
data.clin$OS_STATUS <- as.numeric(data.clin$OS_STATUS)

ix.HPV <- data.clin$`HPV STATUS`=="HPV-"
data.clin <- data.clin[ix.HPV,]
gen.prof <- gen.prof[ix.HPV,]


gen.prof.log <- log2(gen.prof[-which(colnames(gen.prof)=="ID")]+ 0.00001)


ALL <- data.frame("ID"=data.clin$ID,"Time.OS"=data.clin$OS_MONTHS,
                        "Status.OS"=data.clin$OS_STATUS,
                        gen.prof.log[,colnames(gen.prof.log) %in% pemt.genes])

cor.matrix <- rcorr(as.matrix(gen.prof.log[,-ncol(gen.prof.log)]), type = "spearman")
corrplot(cor.matrix$r, 
         method= "shade", type="full", order="hc", tl.cex = .3,
         p.mat = cor.matrix$P, 
         sig.level = 0.05, insig = "blank", tl.col = "black")

cox.sel <- coxph(formula = Surv(time = Time.OS,event = Status.OS) ~ 
                            SLC38A5+CXCL14+P4HA2+C1S, data = ALL)


ALL.expr <- ALL[,colnames(ALL) %in% names(cox.sel$coefficients)]
names <- names(cox.sel$coefficients)
i= 1
for(i in 1:length(names)){
  ALL.expr[,names[i]] <- ALL.expr[,names[i]] * cox.sel$coefficients[names[i]]
  i = i+1
}
rm(names)
# Scale Risk Score (Z-scores)
ALL$RiskScore <- scale(rowSums(ALL.expr))
# Binarize Risk Score based on median
ALL$RiskBin <- c("Risk-","Risk+")[(ALL$RiskScore>median(ALL$RiskScore))+1]
# Add Primary Site to data set and rename factors
ALL$PrimSite <- as.factor(data.clin$`PRIMARY SITE`)
levels(ALL$PrimSite) <- list(Hyp="Hypopharynx",Lar="Larynx",OC="Oral Cavity",Oro="Oropharynx")

# Define Status vector 
Status <- as.factor(ALL$Status.OS)
levels(Status) <- list(LIVING="0",DECEASED="1")

# Quick Explorative Data Analysis of computed Risk Score
rafalib::mypar(2,2)
hist(ALL$RiskScore, main="Histogram of Risk Score", xlab = "Risk Score")
qqnorm(ALL$RiskScore, pch = 1, frame = FALSE, main = "QQ-plot of Risk Score")
qqline(scale(ALL$RiskScore), col = "steelblue", lwd = 2)
wilcox <- wilcox.test(ALL$RiskScore[Status=="LIVING"],ALL$RiskScore[Status=="DECEASED"])
plot(Status,ALL$RiskScore,xlab="",ylab="Risk Score",main=paste0("Wilcox test, p= ",signif(wilcox$p.value,3)))
fisher <- fisher.test(table(Status,ALL$RiskBin))
plot(Status,as.factor(ALL$RiskBin),xlab="",ylab="",main=paste0("Binarized: Fisher's exact p= ",signif(fisher$p.value,3)))
rafalib::mypar(1,1)


cox.RS <- summary(coxph(formula = Surv(time = Time.OS,event = Status.OS) ~ RiskBin,data = ALL))


# Extract values of Cox PH model and display in Kaplan Meier
di.cf <- cox.RS$coefficients
di.ci <-cox.RS$conf.int
hr.di <- round(di.cf[2],2) #hazard ratio
ci.di <- paste(round(di.ci[3],2),"-",round(di.ci[4],2),sep="") #95% CI
fit <- survfit(formula = Surv(time = Time.OS,event = Status.OS) ~ RiskBin,data = ALL)
KM_RiskScore <- survminer::ggsurvplot(fit = fit,conf.int = T,pval = F,risk.table = T, 
                                       legend.labs=c("Risk-","Risk+"), palette = c("chartreuse3","brown3"),
                                       surv.plot.height = 2/3, risk.table.height = 1/3)
KM_RiskScore$plot <- KM_RiskScore$plot+ 
  ggplot2::annotate("text", x = 20, y  = 0.1, # x and y coordinates of the text
                    label = paste("logrank p= ",signif(cox.RS$logtest[3],3),"/ HR:",hr.di,"(95% CI:",ci.di,")", sep = " "), 
                    size = 4)
KM_RiskScore

```

In short, I was able to extract 4 genes from 100 potential candidates and compress them into one simple number: a Risk Score. Then, I used the Risk Score to see if I could group the TCGA patients into low and high Risk (Risk- and Risk+). So, out of 100 genes I could describe 4 genes that do play a role for patient survival and would be candidates for further analysis, e.g. by *in vitro* experiments on functionality. 
The whole analysis with necessary data sets is uploaded to my [GitHub page](https://github.com/HenrikSchinke/RiskScore_pEMT) and a more detailled description is in my [R section on my website](https://henrikschinke.github.io/RiskScore_pEMT.html).

## What do you need to do it yourself?

- A gene signature you expect to impact clinical outcome
- Patient data, e.g. from TCGA
- Basic to intermediate R knowledge
- Basic understanding of Cox models
- GitHub account to download my data & scripts 